### PR TITLE
Feat/autocomplete service examples

### PR DESCRIPTION
**Pull request by BiniCodes**
- Adds an example for the autocomplete service hook
- Update code snippet in README for useAutocompleteService hook

Used a very simple example to have a clean overview, things that can optionally be added later:
- Highlighting matched substring
- Keyboard navigation / handling
- Adding a (x) clear button
- Adding a search icon

Let me know what you think about keeping it very short or adding one of the above 👍

### CODE REVIEW

File: examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+        ref={inputRef}
+        className={cx(
+          styles.searchInput,
+          debouncedInputValue.length && styles.expanded
+        )}
+        value={inputValue}
+        onChange={handleInputChange}
+        autoComplete="off"
+        aria-owns="search-suggestions"
+        aria-expanded={debouncedInputValue.length >= 3}
```
Comment (plumdumpling) [inline examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx:143]: Why `>= 3` here? Line 137 and 152 say it's expanded if `debouncedInputValue.length`. Shouldn't it be the same condition in all places?

File: examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+            styles.suggestions,
+            debouncedInputValue.length && styles.expanded
+          )}
+          id="search-suggestions"
+          role="listbox">
+          {suggestions.map((suggestion, index) => (
+            <li
+              key={suggestion.id}
+              onClick={() => selectSuggestion(suggestion)}
+              id={`search-suggestion-${index}`}
```
Comment (plumdumpling) [inline examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx:160]: Why don't you use the `suggestion.id` here?

File: examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    Array<PlacesAutocompleteServiceSuggestion>
+  >([]);
+  const [suggestionsAreVisible, setSuggestionsAreVisible] =
+    useState<boolean>(false);
+
+  const map = useGoogleMap();
+  const autocompleteService = useAutocompleteService();
+  const placesService = usePlacesService();
+
+  const maxNumberOfSuggestions = 5;
```
[the remark this answers is not in the document]
Comment (BiniCodes) [reply examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx:41]: Sure, do we want to export it?

File: examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx (hunk 1 above)
Comment (BiniCodes) [reply examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx:143]: Actually, I am not using the styles anymore, so I can just remove it :) But good you noticed that here!

File: examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+export interface PlacesAutocompleteServiceSuggestion {
+  id: string;
+  label: string;
+}
+
+const PlacesAutocompleteService: FunctionComponent<
+  Record<string, unknown>
+> = () => {
+  const inputRef = useRef<HTMLInputElement | null>(null);
+  // eslint-disable-next-line no-undef
```
[the remark this answers is not in the document]
Comment (BiniCodes) [reply examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx:26]: Hm, didn't work, yet. But I keep trying 👍

File: examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (plumdumpling) [reply examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx]: Hmm I can't think of anything you need this for, so I think an export isn't needed here.

File: examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (BiniCodes) [reply examples/places-autocomplete-service/components/places-autocomplete-service/places-autocomplete-service.tsx]: check

File: .eslintrc [hunk 5]
```diff
@@ -10,6 +10,14 @@
   "globals": {
     "google": true
   },
```
Comment (BiniCodes) [inline .eslintrc:12]: ` "globals": {
    "google": true
  },`

Can this part be removed now?

**Review by plumdumpling — APPROVED** [review]

**Pull request merged**